File: 3. Arrays/maxProfit.js
/**
 * @param {number[]} prices
 * @return {number}
 */
// 122. Best Time to Buy and Sell Stock II
var maxProfit = function(prices) {
    let maxProfit = 0
    for (let idx = 0; idx < prices.length - 1; idx++){
      if (prices[idx] < prices[idx + 1])
        maxProfit+= prices[idx + 1] - prices[idx]
    }

    return maxProfit;
};

File: 3. Arrays/max_profit.py
# 122. Best Time to Buy and Sell Stock II

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit = 0
        for idx in range(len(prices)- 1):
          if prices[idx + 1] > prices[idx]:
            maxProfit += prices[idx + 1] - prices[idx]
        return maxProfit
